src/luerl.hrl

```diff
@@ -21,95 +21,154 @@
+    tag
+}).
 
 %% Table structure.
--record(tstruct, {data,free,next		%Data, free list, next index
-		 }).
+
+%Data, free list, next index
+-record(tstruct, {data, free, next}).
 
 %% Metatables for atomic datatypes.
 
--record(meta, {nil=nil,
-	       boolean=nil,
-	       number=nil,
-	       string=nil
-	      }).
+-record(meta, {
+    nil = nil,
+    boolean = nil,
+    number = nil,
+    string = nil
+}).
 
 %% Frames for the call stack.
 %% Call return frame
--record(call_frame, {func,args,			%Function, arguments
-		     lvs,			%Local variables
-		     env,			%Environment
-		     is=[],cont=[]		%Instructions, continuation
-		    }).
+
+%Function, arguments
+-record(call_frame, {
+    func,
+    args,
+    %Local variables
+    lvs,
+    %Environment
+    env,
+    %Instructions, continuation
+    is = [],
+    cont = []
+}).
+
 %% Loop break frame
--record(loop_frame, {lvs,			%Local variables
-		     stk,			%Stack
-		     env,			%Environment
-		     is=[],cont=[]		%Instructions, continuation
-		    }).
+
+%Local variables
+-record(loop_frame, {
+    lvs,
+    %Stack
+    stk,
+    %Environment
+    env,
+    %Instructions, continuation
+    is = [],
+    cont = []
+}).
+
 %% Current line
--record(current_line, {line,			%Line
-		       file			%File name
-		      }).
+
+%Line
+-record(current_line, {
+    line,
+    %File name
+    file
+}).
 
 %% Data types.
 
--record(tref, {i}).				%Table reference, index
+%Table reference, index
+-record(tref, {i}).
+
 -define(IS_TREF(T), is_record(T, tref)).
 
--record(table, {a,d=[],meta=nil}).		%Table type, array, dict, meta
+%Table type, array, dict, meta
+-record(table, {a, d = [], meta = nil}).
+
+%Environment reference, index
+-record(eref, {i}).
+
+-define(IS_EREF(E), is_record(E, eref)).
 
--record(eref, {i}).				%Environment reference, index
--define(IS_EREF(E), is_record(E, eref).
+%Userdata reference, index
+-record(usdref, {i}).
 
--record(usdref, {i}).                           %Userdata reference, index
 -define(IS_USDREF(U), is_record(U, usdref)).
 
--record(userdata, {d,meta=nil}).		%Userdata type, data and meta
+%Userdata type, data and meta
+-record(userdata, {d, meta = nil}).
 
--record(thread, {}).				%Thread type
+%Thread type
+-record(thread, {}).
 
 %% There are two function types, the Lua one, and the Erlang one.
 
 %% The environment with upvalues is defined when the function is
 %% referenced and can vary if the function is referenced many
 %% times. Hence it is in the reference not in the the definition.
 
--record(funref, {i,env=[]}).			%Function reference
+%Function reference
+-record(funref, {i, env = []}).
+
 -define(IS_FUNREF(F), is_record(F, funref)).
 
--record(lua_func,{anno=[],			%Annotation
-		  funrefs=[],			%Functions directly referenced
-		  lsz,				%Local var size
-		  %% loc=not_used,		%Local var block template
-		  esz,				%Env var size
-		  %% env=not_used,		%Local env block template
-		  pars,				%Parameter types
-		  b}).				%Code block
+%Annotation
```

Comment:
It seemed to get confused here. I am guessing it was having multiple comments in one line. Not the multiple comment characters but having 2 comments in lines 97 and 99 and ordering broke down. (I am guessing your "comment parser" accepts multiple comment characters as one comment)

Reply:
Aha good catch.